core/src/test/java/org/frankframework/jdbc/JdbcTableListenerTest.java

```diff
@@ -878,4 +877,91 @@ public void testForRaceConditionHandlingOnParallelGet6() throws Exception {
 	public void testForRaceConditionHandlingOnParallelGet7() throws Exception {
 		testForRaceConditionHandlingOnParallelGet(7);
 	}
+
+	@DatabaseTest
+	public void testSelectQueryWithAdditionalFields() throws ConfigurationException {
+		listener.setOrderField("ORDRFLD");
+		listener.setMessageIdField("tINT");
+		listener.setMessageField("tCLOB");
+		listener.setMessageFieldType(JdbcListener.MessageFieldType.CLOB);
+		listener.setAdditionalFields(", tBLOB, tVARCHAR,  ");
+		listener.configure();
+
+		String expected = "SELECT TKEY,tINT,tCLOB,tBLOB,tVARCHAR FROM " + TEST_TABLE + " t WHERE TINT='1' ORDER BY ORDRFLD";
+
+		assertEquals(expected, listener.getSelectQuery());
+		assertEquals(List.of("tBLOB", "tVARCHAR"), listener.getAdditionalFieldsList());
+	}
+
+	@DatabaseTest
+	public void testGetExtraValues() throws Exception {
+		listener.setMessageIdField("tINT");
+		listener.setMessageField("tCLOB");
+		listener.setMessageFieldType(JdbcListener.MessageFieldType.CLOB);
+		listener.setAdditionalFields("tBLOB, tVARCHAR");
+		listener.setBlobsCompressed(false);
+		listener.setBlobSmartGet(false);
+		listener.configure();
+		listener.start();
+
+		try(Connection connection = env.getConnection()) {
+			if (env.getDbmsSupport().getDbms() == Dbms.MSSQL) {
+				JdbcTestUtil.executeStatement(env.getDbmsSupport(), connection, "INSERT INTO " + TEST_TABLE + " (TKEY,TINT,TVARCHAR,TCLOB,TBLOB) VALUES (10,1,'fVC','message',convert(varbinary, 'fBLOB'))", null, new PipeLineSession());
+			} else if (env.getDbmsSupport().getDbms() == Dbms.ORACLE) {
+				JdbcTestUtil.executeStatement(env.getDbmsSupport(), connection, "INSERT INTO " + TEST_TABLE + " (TKEY,TINT,TVARCHAR,TCLOB,TBLOB) VALUES (10,1,'fVC','message',utl_raw.cast_to_raw('fBLOB'))", null, new PipeLineSession());
+			} else {
+				JdbcTestUtil.executeStatement(env.getDbmsSupport(), connection, "INSERT INTO " + TEST_TABLE + " (TKEY,TINT,TVARCHAR,TCLOB,TBLOB) VALUES (10,1,'fVC','message','fBLOB')", null, new PipeLineSession());
+			}
+		}
+
+		PipeLineSession session = new PipeLineSession();
+		RawMessageWrapper<String> rawMessage = listener.getRawMessage(session);
+		Message message = listener.extractMessage(rawMessage, session);
```

Comment:
Ben ook wel benieuwd hoe het in de raw message staat, maar eigenlijk test je hier het packen en unpacken al.

Eigenlijk zou voor de super awesome je hier de raw message moeten serializen en deserializen 🥇 .

Reply:
Maar je test de ADDITIONAL_QUERY_FIELDS_KEY toch? Dat is een Map<String,String> ?